extensions/binding/org.eclipse.smarthome.binding.tradfri/src/main/java/org/eclipse/smarthome/binding/tradfri/internal/model/TradfriVersion.java

```diff
@@ -0,0 +1,78 @@
+/**
+ * Copyright (c) 2014,2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.smarthome.binding.tradfri.internal.model;
+
+import org.eclipse.jdt.annotation.NonNull;
+
+/**
+ * The {@link TradfriVersion} class is a default implementation of comparing TRÅDFRI versions.
+ *
+ * @author Christoph Weitkamp - Initial contribution
+ */
+public class TradfriVersion implements Comparable<TradfriVersion> {
+    private static final String VERSION_PATTERN = "[0-9]+(\\.[0-9]+)*";
+    private static final String VERSION_DELIMITER = "\\.";
+    private final String version;
+
+    /**
+     * Create a new instance.
+     *
+     * @param version the version string
+     */
+    public TradfriVersion(@NonNull String version) {
+        if (version == null)
+            throw new IllegalArgumentException("TradfriVersion cannot be created as version is null.");
+        if (!version.matches(VERSION_PATTERN))
+            throw new IllegalArgumentException("TradfriVersion cannot be created as version has invalid format.");
+        this.version = version;
+    }
+
+    /**
+     * Returns the version string.
+     */
+    public String get() {
+        return version;
+    }
+
+    @Override
+    public int compareTo(@NonNull TradfriVersion other) {
+        if (other == null) {
+            return 1;
+        }
+        String[] thisParts = get().split(VERSION_DELIMITER);
```

Comment:
I would prefer to split the version on object creation once.
If you hold a version reference for multiple comparisons this could speed up the process a little bit.

Reply:
Yes of course. I thought about that too. A StringTokenizer is not an option, am I right?

Reply:
As long as the version uses only integers separated by the version delimiter wouldn't it be much easier to ... in the constructor:
* split the version using the version delimiter into a string array
* create an integer list (e.g. vector) using the same size as the string array
* parse every string part and store it to the correspondent integer part

The compare to could act similar to this one:

```Java
    public static void main(String[] args) {
        System.out.println(compare(Arrays.asList(1, 2, 0, 0), Arrays.asList(1, 2)));
        System.out.println(compare(Arrays.asList(1, 2, 1, 0), Arrays.asList(1, 2)));
        System.out.println(compare(Arrays.asList(1, 2, 0, 0), Arrays.asList(1, 2, 1)));
    }

    public static int compare(List<Integer> myParts, List<Integer> otherParts) {
        final int minPartsSize = Math.min(myParts.size(), otherParts.size());

        for (int i = 0; i < minPartsSize; ++i) {
            final int diff = myParts.get(i) - otherParts.get(i);
            if (diff == 0) {
                continue;
            } else if (diff > 0) {
                return 1;
            } else {
                return -1;
            }
        }

        for (int i = minPartsSize; i < myParts.size(); ++i) {
            if (myParts.get(i) != 0) {
                return 1;
            }
        }

        for (int i = minPartsSize; i < otherParts.size(); ++i) {
            if (otherParts.get(i) != 0) {
                return -1;
            }
        }

        return 0;
    }
```